frontend/src/metabase/query_builder/components/expressions/ExpressionEditorTextfield/helper-text-strings.ts

```diff
@@ -799,8 +805,28 @@ See the full list here: https://w.wiki/4Jx`,
   },
 ];
 
-export const getHelpText = (name: string): HelpText | undefined => {
-  return helperTextStrings.find(h => h.name === name);
+export const getHelpText = (
+  name: string,
+  database: Database,
+): HelpText | undefined => {
+  const helperText = helperTextStrings.find(h => h.name === name);
+
+  if (helperText && name === "now") {
+    helperText.description = getDescriptionForNow(database);
+  }
+
+  return helperText;
+};
+
+const getNowAtTimezone = (timezone: string) =>
+  timezone ? moment().tz(reportTimezone).format("LT") : moment().format("LT");
+
+const getDescriptionForNow = (database: Database) => {
+  const hasTimezoneFeatureFlag = database.features.includes("set-timezone");
+  const timezone = hasTimezoneFeatureFlag ? reportTimezone : "UTC";
```

Comment:
Unfortunately I just realised this logic won't work for H2, which is special and displays `now` in the system time zone, regardless of the reportTimezone.
We could either:
1. Show a different description specifically for H2, which leaves out the now "Currently ${nowAtTimezone} in ${timezone}".
2. Get the system time zone in the FE (either async or in settings) and display it here.
3. Give up and do what we maybe should have done initially, which was return the now time directly with an API request and display it asynchronously.

Given report time zones are getting a shake-up in 46 https://github.com/metabase/metabase/issues/27178 and H2 is hardly used except for the sample database, I'd lean towards 1 for now. @gusaiani What do you think?